executor/adapter.go

```diff
@@ -295,8 +296,12 @@ func (a *ExecStmt) PointGet(ctx context.Context) (*recordSet, error) {
 	}
 	a.Ctx.GetSessionVars().StmtCtx.Priority = kv.PriorityHigh
 
+	var pointExecutor *PointGetExecutor
+	useMaxTS := startTs == math.MaxUint64
+
 	// try to reuse point get executor
-	if a.PsStmt.Executor != nil {
+	// We should only use the cached the executor when the startTS is MaxUint64
+	if a.PsStmt.Executor != nil && useMaxTS {
```

Comment:
Could you add some comments around [the definition of `PlanCacheStmt.Executor`](https://github.com/pingcap/tidb/blob/master/planner/core/plan_cache_utils.go#L412) as well to indicate there is a dangerous assumption that the cached Executor always uses `MaxTS`?

Reply:
Done